File: AITutor/AI_Tutor/AI_Tutor/Models/Problem.swift
// AI_Tutor/Models/Problem.swift
import Foundation
import Observation

@Observable
class Problem: Identifiable {
    let id = UUID()
    var number: Int
    var fullLatexText: String
    var knownDataMarkdown: String
    var isSelected: Bool = true

    init(number: Int, fullLatexText: String, knownDataMarkdown: String) {
        self.number = number
        self.fullLatexText = fullLatexText
        self.knownDataMarkdown = knownDataMarkdown
    }
}

File: AITutor/app/src/main/java/com/ai/aitutorandroid/models/Problem.kt
package com.ai.aitutorandroid.models

import java.util.UUID

data class Problem(
    val id: String = UUID.randomUUID().toString(),
    val number: Int,
    val fullLatexText: String,
    val knownDataMarkdown: String,
    val isSelected: Boolean = true
)
